In [2]:
import numpy as np
import pandas as pd
from tqdm.notebook import tqdm
import ast


In [2]:
import sys
sys.path.append('./src')
# from dataloader import create_dataloaders
from evaluation import evaluation

In [9]:
# test = pd.read_csv('train/router/train_output_MoE.csv',index_col=0)
input = test[test['dataset']=='abt-buy']

In [10]:
evaluator = evaluation()
evaluator.process('ER','wdc',input)

Precision:0.583941605839416

Recall:0.3883495145631068

F1:0.46647230320699706


## ER Evaluation Task

In [ ]:
### Calculate the size of dataset

In [5]:
# train = pd.read_csv('/data/home/wangys/MoE-Example/DI/restaurant/restaurant_all.csv')
train = pd.read_json('/data/home/wangys/ExtractGPT-main/data/processed_datasets/oa-mine/train_1.0.jsonl',lines=True)
len(train)

1452

In [24]:
train

input category  \
0    Vertvie Women Running Shorts Trunks Phone Pock...   Shorts   
1    New Running Sport Mens Basketball Tight Compre...   Shorts   
2    Space Jam Tune Squad Men&#39;s Basketball Shor...   Shorts   
3    PEAK Women Running Shorts Yoga Drawstring Shor...   Shorts   
4    PENERAN 2018 Sport Shorts Women Yoga Fitness S...   Shorts   
..                                                 ...      ...   
306  FreeBee Windproof Unisex Cycling Goggles Sport...  Eyewear   
307  100 speedcraft Base Outdoor Sports Bicycle Sun...  Eyewear   
308  Professional Ski Goggles UV400 Lens Anti-fog S...  Eyewear   
309  Cycling Eyewear Unisex Outdoor Sunglass UV400 ...  Eyewear   
310  POLISI Professional Ski Goggles Double Layer L...  Eyewear   

                                         target_scores  
0    {'Sport Type': {'Running': 1}, 'Gender': {'Wom...  
1    {'Sport Type': {'Running': 1}, 'Gender': {'Men...  
2    {'Sport Type': {'Basketball': 1}, 'Gender': {'...  
3    {'Brand Name': {'PEAK': 1}, 'Gender': {'Women'...  
4    {'Gender': {'Women': 1}, 'Brand Name': {'PENER...  
..                                                 ...  
306  {'Gender': {'Unisex': 1}, 'Sport Type': {'Cycl...  
307  {'Lenses Material': {'n/a': 1}, 'Sport Type': ...  
308  {'Lenses Optical Attribute': {'n/a': 1}, 'Gend...  
309  {'Lenses Optical Attribute': {'Polarized': 1},...  
310  {'Sport Type': {'Skiing': 1}, 'Frame Material'...  

[311 rows x 3 columns]

In [14]:
array = np.load('/data/home/wangys/LLM_ER/process_data/semi-text-w/semi-text-w.npy',allow_pickle=True)[0]
array

[(4809, 4809),
 (2500, 2500),
 (3917, 3917),
 (4928, 4928),
 (2700, 2700),
 (5307, 5307),
 (556, 556),
 (4660, 4660),
 (3653, 3653),
 (2910, 2910),
 (108, 108),
 (3994, 3994),
 (2868, 2868),
 (4221, 4221),
 (5076, 5076),
 (4799, 4799),
 (5235, 5235),
 (351, 351),
 (2463, 2463),
 (366, 366),
 (3408, 3408),
 (2675, 2675),
 (3756, 3756),
 (622, 622),
 (3864, 3864),
 (2467, 2467),
 (4089, 4089),
 (1550, 1550),
 (127, 127),
 (478, 478),
 (2118, 2118),
 (4464, 4464),
 (3896, 3896),
 (652, 652),
 (1058, 1058),
 (985, 985),
 (5185, 5185),
 (3606, 3606),
 (4191, 4191),
 (2565, 2565),
 (291, 291),
 (4977, 4977),
 (2992, 2992),
 (4047, 4047),
 (2383, 2383),
 (33, 33),
 (4001, 4001),
 (1236, 1236),
 (257, 257),
 (244, 244),
 (3905, 3905),
 (2989, 2989),
 (4987, 4987),
 (3914, 3914),
 (2621, 2621),
 (863, 863),
 (987, 987),
 (5316, 5316),
 (1691, 1691),
 (936, 936),
 (3831, 3831),
 (3705, 3705),
 (4145, 4145),
 (5208, 5208),
 (1179, 1179),
 (2509, 2509),
 (4026, 4026),
 (1475, 1475),
 (966, 966),
 

In [29]:
def list_files_in_folder(folder_path):
    """
    List all files in the given folder and its subfolders, and return their paths.

    Parameters:
    folder_path (str): The path of the folder to traverse.

    Returns:
    list: A list of file paths
    """
    file_paths = []
    for root, dirs, files in os.walk(folder_path):
        for file in files:
            file_path = os.path.join(root, file)
            file_paths.append(file_path)
    
    return file_paths

In [67]:
from sklearn.metrics import precision_score,recall_score,f1_score

In [68]:
# result_merge = pd.read_csv('/data/home/wangys/LLaMA-Factory-main/inference/Mistral|synthea_train-MoE-Add/mistral-7b-synthea_test_few_output.csv',index_col=0).fillna('')
result_merge = pd.read_csv('/data/home/wangys/LLaMA-Factory-main/inference_MELD/Vicuna-33B|amazon-google-train-MoE/vicuna-33b-MoE_ER_amazon-google-test.csv',index_col=0).fillna('')
def Transfer(row):
    if(row['output'].__contains__('dismatch')):
        label = 0
    else:
        label = 1
    if(row['predict'].__contains__('dismatch')):
        predict = 0
    else:
        predict = 1
    return label,predict
result_output = result_merge.apply(Transfer,axis=1,result_type='expand')
from sklearn.metrics import f1_score
print(precision_score(y_true=result_output[0],y_pred=result_output[1]),recall_score(y_true=result_output[0],y_pred=result_output[1]),f1_score(y_true=result_output[0],y_pred=result_output[1]))

0.03225806451612903 0.004273504273504274 0.007547169811320755


In [80]:
result_merge[result_merge['predict'].str.strip()!=result_merge['output']]['output'].value_counts()

{'Output': 'match'}       16
{'Output': 'dismatch'}    15
Name: output, dtype: int64

## Amazon-Google
Mistral: 0.8337
LLaMa2-13b: 0.8086956521739128

## ant_buy_add
0.5714285714285714 amazon-google-test-DPO-AllPrompt.csv
0.8925233644859812 ant-buy-test-sft-align.csv
0.8750000000000001 beer-prompt-init-test.csv
0.0 dblp-acm-PromptOpt-test.csv
0.0 dblp-google-PromptOpt-test.csv
0.8055555555555556 semi-text-c-prompt-init-test.csv
0.7014492753623189 semi-text-w-prompt-init-test-all.csv
0.8169642857142858 walmart-amazon-PromptOpt-test-align.csv
0.8883534136546185 wdc-all-PromptOpt-test.csv
## semi_text_c
0.4602683178534572 amazon-google-test-DPO-AllPrompt.csv
0.8189655172413793 ant-buy-test-sft-align.csv
0.8235294117647058 beer-prompt-init-test.csv
0.030973451327433624 dblp-acm-PromptOpt-test.csv
0.00558659217877095 dblp-google-PromptOpt-test.csv
0.8124392614188533 semi-text-c-prompt-init-test.csv
0.7268170426065163 semi-text-w-prompt-init-test-all.csv
0.6123778501628665 walmart-amazon-PromptOpt-test-align.csv
0.857037037037037 wdc-all-PromptOpt-test.csv
## walmart_amazon_add
0.4689092762487258 amazon-google-test-DPO-AllPrompt.csv
0.8016194331983806 ant-buy-test-sft-align.csv
0.8750000000000001 beer-prompt-init-test.csv
0.0 dblp-acm-PromptOpt-test.csv
0.32088768930874195 dblp-google-PromptOpt-test.csv
0.8588235294117647 semi-text-c-prompt-init-test.csv
0.7438752783964365 semi-text-w-prompt-init-test-all.csv
0.6904315196998123 walmart-amazon-PromptOpt-test-align.csv
0.8662232076866222 wdc-all-PromptOpt-test.csv
## wdc_all_add_llama2_13b
0.5638297872340425 amazon-google-test-DPO-AllPrompt.csv
0.86784140969163 ant-buy-test-sft-align.csv
0.8750000000000001 beer-prompt-init-test.csv
0.0 dblp-acm-PromptOpt-test.csv
0.0 dblp-google-PromptOpt-test.csv
0.8140900195694716 semi-text-c-prompt-init-test.csv
0.7298050139275765 semi-text-w-prompt-init-test-all.csv
0.7540983606557377 walmart-amazon-PromptOpt-test-align.csv
0.8932190179267342 wdc-all-PromptOpt-test.csv
## Mixtral_8*7b ant_buy
0.6156716417910447 amazon-google-test-DPO-AllPrompt.csv
0.852607709750567 ant-buy-test-sft-align.csv
0.9032258064516129 beer-prompt-init-test.csv
0.6023113528212102 dblp-acm-PromptOpt-test.csv
0.8178311187312474 dblp-google-PromptOpt-test.csv
0.8198019801980199 semi-text-c-prompt-init-test.csv
0.7542857142857142 semi-text-w-prompt-init-test-all.csv
0.8017817371937639 walmart-amazon-PromptOpt-test-align.csv
0.8900116595413914 wdc-all-PromptOpt-test.csv

In [68]:
import os
file_list = list_files_in_folder('/data/home/wangys/LLaMA-Factory-main/inference/Mistral|wdc_all-MoE-Add')
for f in file_list:
    result_merge = pd.read_csv(f,index_col=0).fillna('')
    def Transfer(row):
        if(row['output'].__contains__('dismatch')):
            label = 0
        else:
            label = 1
        if(row['predict'].__contains__('dismatch')):
            predict = 0
        else:
            predict = 1
        return label,predict
    result_output = result_merge.apply(Transfer,axis=1,result_type='expand')
    from sklearn.metrics import f1_score
    print(f1_score(y_true=result_output[0],y_pred=result_output[1]),f.split('/')[-1].replace('mistral-7b-',''))

0.6331168831168832 amazon-google-test.csv
0.8306878306878307 ant_buy_test_output.csv
0.8241010689990281 semi-text-c-test-MoE.csv
0.7643979057591623 semi-text-w-test-MoE.csv
0.8298368298368298 walmart_amazon_test_output.csv
0.9038219235615288 wdc_all_test_output.csv
1.0 RE-test_t=4.csv


In [ ]:
amazon_google
0.8341013824884792 amazon-google-test.csv
0.7537091988130563 ant_buy_test_output.csv
0.7467467467467468 semi-text-c-test-MoE.csv
0.6578947368421052 semi-text-w-test-MoE.csv
0.6635071090047393 walmart_amazon_test_output.csv
0.8806278397356464 wdc_all_test_output.csv
1.0 RE-test_t=4.csv
ant_buy
0.6884681583476764 amazon-google-test.csv
0.9112709832134294 ant_buy_test_output.csv
0.8744512730465319 semi-text-c-test-MoE.csv
0.7828282828282829 semi-text-w-test-MoE.csv
0.8367816091954023 walmart_amazon_test_output.csv
0.9197816043679127 wdc_all_test_output.csv
semi_text_c
0.5965909090909091 amazon-google-test.csv
0.8258426966292136 ant_buy_test_output.csv
0.7942973523421588 semi-text-c-test-MoE.csv
0.7272727272727272 semi-text-w-test-MoE.csv
0.8555240793201133 walmart_amazon_test_output.csv
0.8821670428893904 wdc_all_test_output.csv
semi_text_w
0.731404958677686 amazon-google-test.csv
0.8888888888888888 ant_buy_test_output.csv
0.8646125116713352 semi-text-c-test-MoE.csv
0.7616580310880829 semi-text-w-test-MoE.csv
0.8542713567839197 walmart_amazon_test_output.csv
0.901169826542961 wdc_all_test_output.csv
walmart_amazon
0.6715063520871144 amazon-google-test.csv
0.918781725888325 ant_buy_test_output.csv
0.8421052631578948 semi-text-c-test-MoE.csv
0.7688172043010753 semi-text-w-test-MoE.csv
0.9142857142857143 walmart_amazon_test_output.csv
0.8979947689625108 wdc_all_test_output.csv
wdc_all
0.6331168831168832 amazon-google-test.csv
0.8306878306878307 ant_buy_test_output.csv
0.8241010689990281 semi-text-c-test-MoE.csv
0.7643979057591623 semi-text-w-test-MoE.csv
0.8298368298368298 walmart_amazon_test_output.csv
0.9038219235615288 wdc_all_test_output.csv

In [ ]:
## Evaluation on AVE, whether it can be done or not?

In [47]:
ave_train = pd.read_csv('/data/home/wangys/LLaMA-Factory-main/inference/llama-2-13b-lora_weight|oa_mine|oa_mine_train_small-oa_mine_test_small.csv',index_col=0)

In [154]:
ave_train = pd.read_csv('inference_Transfer_ER/select--oa_mine-test.csv',index_col=0)

In [155]:
import ast
def AST(row):
    output = row['output'].strip()
    predict = row['predict'].strip()
    output_item = list(ast.literal_eval(output).values())[0]
    predict_item = list(ast.literal_eval(predict).values())[0]
    row['output_item'] = output_item.lower()
    row['predict_item'] = predict_item.lower()
    return row
ave_train = ave_train.apply(AST,axis=1)

In [156]:
1 - len(ave_train[ave_train['output_item']!=ave_train['predict_item']]) / len(ave_train)

0.7405140758873929

In [51]:
ave_train[ave_train['output_item']!=ave_train['predict_item']]

instruction  input  \
5     You are a world-class expert for extracting in...    NaN   
8     You are a world-class expert for extracting in...    NaN   
12    You are a world-class expert for extracting in...    NaN   
18    You are a world-class expert for extracting in...    NaN   
24    You are a world-class expert for extracting in...    NaN   
...                                                 ...    ...   
2421  You are a world-class expert for extracting in...    NaN   
2423  You are a world-class expert for extracting in...    NaN   
2432  You are a world-class expert for extracting in...    NaN   
2440  You are a world-class expert for extracting in...    NaN   
2447  You are a world-class expert for extracting in...    NaN   

                                                 output  \
5                           {"Brand": "Glamorous Wash"}   
8            {"Specific uses": "Laundry Scent Booster"}   
12                            {"Brand": "Arm & Hammer"}   
18                        {"Specialty": "Enzyme-Based"}   
24                           {"Specialty": "Signature"}   
...                                                 ...   
2421                           {"Item form": "Tablets"}   
2423  {"Health benefit": "Superior Strength Hair/Ski...   
2432    {"Supplement type": "Pure Quercetin Dihydrate"}   
2440     {"Administration type": "Sublingual Vitamins"}   
2447  {"Supplement type": " Brewers Yeast 7-1/2 Grai...   

                                                predict  \
5                          {"Brand": "Tyler Candle Co"}   
8             {"Specific uses": "Cruelty Free Formula"}   
12             {"Brand": "Arm & Hammer Clean & Simple"}   
18                        {"Specialty": "Free & Clear"}   
24                             {"Specialty": "Classic"}   
...                                                 ...   
2421             {"Item form": "Tablets w. Folic Acid"}   
2423   {"Health benefit": "Hair/Skin/Nails with Arga...   
2432         {"Supplement type": "Quercetin Dihydrate"}   
2440              {"Administration type": "Sublingual"}   
2447   {"Supplement type": "Brewers Yeast 7-1/2 Grai...   

                                            output_item  \
5                                        glamorous wash   
8                                 laundry scent booster   
12                                         arm & hammer   
18                                         enzyme-based   
24                                            signature   
...                                                 ...   
2421                                            tablets   
2423                  superior strength hair/skin/nails   
2432                           pure quercetin dihydrate   
2440                                sublingual vitamins   
2447   brewers yeast 7-1/2 grains with vitamin b1, v...   

                                           predict_item  
5                                       tyler candle co  
8                                  cruelty free formula  
12                          arm & hammer clean & simple  
18                                         free & clear  
24                                              classic  
...                                                 ...  
2421                              tablets w. folic acid  
2423    hair/skin/nails with argan/coconut oil/collagen  
2432                                quercetin dihydrate  
2440                                         sublingual  
2447  brewers yeast 7-1/2 grains with vitamin b1, vi...  

[622 rows x 6 columns]

In [ ]:
0.8049932523616734 ave_test_large
0.7442645074224021 ave_test_small
0.8049775601795186 oa_mine_large
0.7462260301917585 oa_mine_small

## Evaluation on DI
### Walmart

In [145]:
# walmart_DI_result = pd.read_csv('/data/home/wangys/LLaMA-Factory-main/inference/Mistral|DI-amazon-train-all/mistral-7b-amazon_test_output_wide.csv',index_col=0)
# walmart_DI_result = pd.read_csv('/data/home/wangys/LLaMA-Factory-main/inference_baseline/models|jellyfish-sft/llama2-13b-walmart_test.csv',index_col=0)
# walmart_DI_result = pd.read_csv('/data/home/wangys/LLaMA-Factory-main/inference_baseline/models|jellyfish-sft/llama2-13b-amazon_test.csv',index_col=0)
# walmart_DI_result = pd.read_csv('/data/home/wangys/LLaMA-Factory-main/inference_baseline/models|Mixtral-sft/Mixtral-walmart_test.csv',index_col=0)
# walmart_DI_result = pd.read_csv('/data/home/wangys/LLaMA-Factory-main/inference_baseline/models|Mixtral-sft/Mixtral-amazon_test.csv',index_col=0)
# walmart_DI_result = pd.read_csv('/data/home/wangys/LLaMA-Factory-main/inference_baseline/models|Mixtral-sft/Mixtral-restaurant_test.csv',index_col=0)
walmart_DI_result = pd.read_csv('/data/home/wangys/LLaMA-Factory-main/inference/Mistral|DI-amazon-train-all/mistral-7b-amazon_test_output_wide.csv',index_col=0)

In [ ]:
walmart_DI_result = pd.read_csv('/data/home/wangys/LLaMA-Factory-main/inference/Mistral|DI-amazon-train-all/mistral-7b-amazon_test_output_wide.csv',index_col=0)

In [146]:
# walmart_DI_result[walmart_DI_result['output']!=walmart_DI_result['predict']]
import ast
def AST(row):
    output = list(ast.literal_eval(row['output']).values())[0]
    try:
        predict = list(ast.literal_eval(row['predict'].strip()).values())[0]
    except:
        print(row['predict'])
        predict  = ''
    return output,predict
walmart_DI_ast = walmart_DI_result.apply(AST,axis=1,result_type='expand')

In [147]:
1 - len(walmart_DI_ast[walmart_DI_ast[0]!=walmart_DI_ast[1]]) / len(walmart_DI_ast)

0.7512254901960784

In [141]:
len(walmart_DI_ast),len(walmart_DI_ast[walmart_DI_ast[0]!=walmart_DI_ast[1]])

(2451, 555)

In [26]:
1 - 2/29

0.9310344827586207

In [3]:
walmart_test = pd.read_csv('/data/home/wangys/LLaMA-Factory-main/inference/Mistral|restaurant_train-MoE-Add/mistral-7b-restaurant_test_output_wide.csv',index_col=0)

In [142]:
walmart_test = pd.read_csv('/data/home/wangys/LLaMA-Factory-main/inference/Mistral|DI-amazon-train-all/mistral-7b-amazon_test_output_wide.csv',index_col=0)

In [151]:
# walmart_test = pd.read_csv('/data/home/wangys/LLaMA-Factory-main/inference/CTA|CTA_SimTab_train_init/mistral-7b-SimTab_test_few.csv',index_col=0)
# walmart_test = pd.read_csv('/data/home/wangys/LLaMA-Factory-main/inference_baseline/models|jellyfish-sft/llama2-13b-restaurant_test.csv',index_col=0)
# walmart_test = pd.read_csv('inference_Transfer_ER/select--walmart-test.csv',index_col=0)
walmart_test = pd.read_csv('inference_Transfer_ER/select--amazon-test.csv',index_col=0)

In [152]:
# walmart_DI_result[walmart_DI_result['output']!=walmart_DI_result['predict']]
import ast
def AST(row):
    output = list(ast.literal_eval(row['output']).values())[0]
    try:
        predict = list(ast.literal_eval(row['predict'].strip()).values())[0]
    except:
        print(row['predict'])
        predict  = ''
    return output,predict
walmart_DI_ast = walmart_test.apply(AST,axis=1,result_type='expand')


In [153]:
1 - len(walmart_DI_ast[walmart_DI_ast[0]!=walmart_DI_ast[1]]) / len(walmart_DI_ast)

0.6678921568627452

{'init_select': 9.400069952011108, 'init-train': 202.6307315826416, 'batch-division': 210.34611916542053, 'gradient-calculation': 256.1557722091675, 'IF-Score': 282.9054231643677, 'Final Selection': 283.1132471561432, 'Selection fine-tune': 860.2257125377655, 'Selection Inference': 904.9117484092712}

### Evaluation Over CTA

In [132]:
SimTab_test = pd.read_csv('/data/home/wangys/LLaMA-Factory-main/inference/CTA|CTA_SimTab_train_init/mistral-7b-SimTab_test_few.csv',index_col=0)

In [31]:
SimTab_test = pd.read_csv('/data/home/wangys/LLaMA-Factory-main/inference/CTA|CTA_WebTable_train_init/mistral-7b-WebTable_Test_few.csv',index_col=0)

In [2]:
SimTab_test = pd.read_csv('/data/home/wangys/LLaMA-Factory-main/inference_GEIL/Mistral|RE-MoE-Add/mistral-7b-data_RE_RE-test_t=4.csv',index_col=0)

In [38]:
SimTab_test = pd.read_csv('inference_Transfer_ER/--RE-test.csv',index_col=0)

In [64]:
SimTab_test = pd.read_csv('/data/home/wangys/LLaMA-Factory-main/inference_MELD/Vicuna-33B|RE-MoE-Add/vicuna-33b-data_RE_RE-test_t=4.csv',index_col=0)

In [117]:
# SimTab_test = pd.read_csv('/data/home/wangys/LLaMA-Factory-main/inference_baseline/models|Mixtral-sft/Mixtral-SimTab_test.csv',index_col=0)
SimTab_test = pd.read_csv('/data/home/wangys/LLaMA-Factory-main/inference_baseline/models|Mixtral-sft/Mixtral-webtable_test.csv',index_col=0)

In [57]:
SimTab_test = pd.read_csv('/data/home/wangys/LLaMA-Factory-main/inference_MELD/Vicuna-33B|SimTab_train_init-MoE-Add/vicuna-33b-data_CTA_SimTab_test_few.csv',index_col=0)

In [84]:
SimTab_test = pd.read_csv('/data/home/wangys/LLaMA-Factory-main/inference_GEIL/Mistral|SimTab-MoE-CT/mistral-7b-data_CTA_SimTab_test_few.csv',index_col=0)

In [51]:
SimTab_test = pd.read_csv('/data/home/wangys/LLaMA-Factory-main/inference_baseline/models|Mixtral-sft/Mixtral-RE-test.csv',index_col=0)

In [19]:
SimTab_test = pd.read_csv('/data/home/wangys/LLaMA-Factory-main/inference_baseline/models|table-llama/table-llama-SimTab_test.csv',index_col=0)

In [35]:
SimTab_test = pd.read_csv('/data/home/wangys/LLaMA-Factory-main/inference_baseline/models|jellyfish-sft/table-llama-webtable_test.csv',index_col=0)

In [49]:
SimTab_test = pd.read_csv('/data/home/wangys/LLaMA-Factory-main/inference_baseline/models|table-llama/table-llama-RE-test.csv',index_col=0)

In [21]:
SimTab_test = pd.read_csv('inference_Transfer_ER/SimTab-P1--SimTab_test_few.csv',index_col=0)

In [121]:
SimTab_test = pd.read_csv('inference_Transfer_ER/--SimTab_test_few.csv',index_col=0)

In [70]:
SimTab_test = pd.read_csv('inference_Transfer_ER/--WebTable_Test_few.csv',index_col=0)

In [128]:
## Only for RE
SimTab_test['index'] = SimTab_test.index

In [133]:
# SimTab_test
import ast
def Ast(row):
    truth = list(eval(row['output']).values())[0]
    try:
        pred = list(eval(row['predict']).values())[0]
    except:
        pred = ''
    index = row['index']
    return truth, pred, index
SimTab_test_Transform = SimTab_test.apply(Ast,axis=1,result_type='expand')

In [6]:
all_relation = list(SimTab_test_Transform[0].unique())

In [123]:
all_relation = np.load('/data/home/wangys/LLaMA-Factory-main/data/CTA/sim_all_relation.npy',allow_pickle=True)

In [72]:
all_relation = np.load('/data/home/wangys/LLaMA-Factory-main/data/CTA/webtable_all_relation.npy',allow_pickle=True)

In [67]:
all_relation = np.load('/data/home/wangys/LLaMA-Factory-main/data/RE/all_relation.npy',allow_pickle=True)

## Calculate coverage for multiple answers

In [2]:
import pandas as pd
RE_test = pd.read_csv('train/RE/test_RAG.csv',index_col=0)


In [3]:
RE_test

query  \
0     <table_title>yukon legislative assemblycurrent...   
1     <table_title>list of songs recorded by led zep...   
2     <table_title>list of air india fc managersstat...   
3            <table_title>astanasport<header>club|sport   
4           <table_title>astanasport<header>club|league   
...                                                 ...   
2067  <table_title>list of rulers of austriahouse of...   
2068  <table_title>list of rulers of austriahouse of...   
2069  <table_title>list of rulers of austriahouse of...   
2070  <table_title>list of rulers of austriahouse of...   
2071  <table_title>list of rulers of austriahouse of...   

                                                    pos  \
0     ['government.politician.party-government.polit...   
1                        ['music.composition.composer']   
2     ['people.person.nationality', 'people.person.p...   
3                          ['sports.sports_team.sport']   
4     ['sports.sports_team.league-sports.sports_leag...   
...                                                 ...   
2067  ['people.person.spouse_s-people.marriage.spouse']   
2068          ['people.deceased_person.place_of_death']   
2069                          ['people.person.parents']   
2070  ['people.person.spouse_s-people.marriage.spouse']   
2071          ['people.deceased_person.place_of_death']   

                                                    neg  index  \
0     ['time.event.locations', 'base.wikipedia_infob...      0   
1     ['time.event.locations', 'base.wikipedia_infob...      1   
2     ['time.event.locations', 'base.wikipedia_infob...      2   
3     ['time.event.locations', 'base.wikipedia_infob...      3   
4     ['time.event.locations', 'base.wikipedia_infob...      4   
...                                                 ...    ...   
2067  ['time.event.locations', 'base.wikipedia_infob...   2067   
2068  ['time.event.locations', 'base.wikipedia_infob...   2068   
2069  ['time.event.locations', 'base.wikipedia_infob...   2069   
2070  ['time.event.locations', 'base.wikipedia_infob...   2070   
2071  ['time.event.locations', 'base.wikipedia_infob...   2071   

                                                   pred  \
0     government.politician.party-government.politic...   
1                            music.composition.composer   
2                             people.person.nationality   
3                              sports.sports_team.sport   
4     sports.sports_team.league-sports.sports_league...   
...                                                 ...   
2067      people.person.spouse_s-people.marriage.spouse   
2068              people.deceased_person.place_of_death   
2069                       people.person.place_of_birth   
2070      people.person.spouse_s-people.marriage.spouse   
2071              people.deceased_person.place_of_death   

                                                  truth  \
0     government.politician.party-government.politic...   
1                            music.composition.composer   
2                             people.person.nationality   
3                              sports.sports_team.sport   
4     sports.sports_team.league-sports.sports_league...   
...                                                 ...   
2067      people.person.spouse_s-people.marriage.spouse   
2068              people.deceased_person.place_of_death   
2069                              people.person.parents   
2070      people.person.spouse_s-people.marriage.spouse   
2071              people.deceased_person.place_of_death   

                                                  top-k  \
0     ['government.politician.party-government.polit...   
1     ['music.composition.composer', 'award.award_wi...   
2     ['people.person.nationality', 'people.person.p...   
3     ['sports.sports_team.sport', 'sports.sports_te...   
4     ['sports.sports_team.league-sports.sports_leag...   
...                                                 ...   
2067  ['

: 

In [11]:
SimTab_test_Transform.columns = ['truth','pred','index']
count = 0
truth_list = []
pred_list = []
for i in range(len(SimTab_test_Transform['index'].unique())):
    select_df = SimTab_test_Transform[SimTab_test_Transform['index'] == i]
    # truth = select_df.iloc[0,0]
    truth = eval(RE_test.iloc[i,1])
    select_df_filter = select_df[select_df['pred'].isin(all_relation)]
    try:
        pred = select_df_filter['pred'].value_counts().idxmax()
    except:
        pred = select_df['pred'].value_counts().idxmax()
    truth_list.append(truth)
    pred_list.append(pred)
    if truth.__contains__(pred):
        count += 1
    # else:
    #     print(truth,select_df['pred'].value_counts().idxmax())
count / len(SimTab_test_Transform['index'].unique())

NameError: name 'SimTab_test_Transform' is not defined

## SimTab/WebTable

In [134]:
SimTab_test_Transform.columns = ['truth','pred','index']
count = 0
truth_list = []
pred_list = []
for i in range(len(SimTab_test_Transform['index'].unique())):
    select_df = SimTab_test_Transform[SimTab_test_Transform['index'] == i]
    truth = select_df.iloc[0,0]
    select_df_filter = select_df[select_df['pred'].isin(all_relation)]
    try:
        pred = select_df_filter['pred'].value_counts().idxmax()
    except:
        pred = select_df['pred'].value_counts().idxmax()
    truth_list.append(truth)
    pred_list.append(pred)
    if truth==pred:
        count += 1
    # else:
    #     print(truth,select_df['pred'].value_counts().idxmax())
count / len(SimTab_test_Transform['index'].unique())

0.8935611038107752

In [135]:
relation_dict = {}
# all_relation = list(SimTab_test_Transform['truth'].unique())
# all_relation = np.load('data/CTA/sim_all_relation.npy')
# all_relation.extend(list(SimTab_test_Transform['pred'].unique()))
for i in range(len(all_relation)):
    relation_dict[all_relation[i]] = i

In [136]:
SimTab_F1 = pd.DataFrame()
SimTab_F1['pred_output'] = pred_list
SimTab_F1['truth_output'] = truth_list
from sklearn.metrics import f1_score
pred = SimTab_F1['pred_output'].map(relation_dict).fillna(0).to_list()
truth = SimTab_F1['truth_output'].map(relation_dict).to_list()
f1_score(y_pred=pred,y_true=truth,average='micro'),f1_score(y_pred=pred,y_true=truth,average='macro')

(0.8935611038107752, 0.761725380779237)

In [ ]:
(0.832046332046332, 0.6213512764551432) RE on Full-Size?
(0.8127413127413128, 0.63080375905545) RE select 2k
(0.8359073359073359, 0.6677882789061006) RE-Full
0.8972007722007722 RE-Full Coverage
0.9005791505791506 RE-Select Coverage
0.7398648648648649 Mistral-Coverage

(0.8909329829172142, 0.771330102155335) Mistral
(0.7818659658344284, 0.5989916556768511) Qwen
(0.859395532194481, 0.731767797493626) Qwen
(0.8528252299605782, 0.780033122507523) Select-Mistral-2K sample
(0.8462549277266754, 0.7313185684839995) 
Select-Mistral-1K sample

In [ ]:
from sklearn.metrics import f1_score
pred = SimTab_F1['pred_output'].map(relation_dict).fillna(0).to_list()
truth = SimTab_F1['truth_output'].map(relation_dict).to_list()
f1_score(y_pred=pred,y_true=truth,average='micro')

In [ ]:
(0.8625492772667542, 0.6750286265992705) Mistral-Macro
(0.961488508667909, 0.7905716732210353) Mistral-Macro
(0.8222076215505915, 0.5598983999723887) TableLLAMA-SimTab_F1
(0.9459596815178722, 0.8095254469231098) TableLLAMA
(0.832046332046332, 0.6213512764551432) RE-test Mistral-7B
(0.6587837837837838, 0.4239049298280863) TableLLAMA-RE

WebTable
(0.8783701188455009, 0.678877623513516) DataSelection 1500
(0.9046519524617996, 0.7002269601639742) 2000


(0.9403735144312394, 0.7607254907599078) DataSelection 4000/14000


In [ ]:
from sklearn.metrics import f1_score
pred = SimTab_test_Transform['pred'].map(relation_dict).to_list()
truth = SimTab_test_Transform['truth'].map(relation_dict).to_list()
f1_score(y_pred=pred,y_true=truth,average='macro')

In [ ]:
# 0.8935611038107752
# 0.8625492772667542 Micro-F1
# 0.7891529869661876/0.6750286265992705 Macro-F1
0.8935 Micro/ 0.7616 Macro
0.9641 Micro/ 0.7876 Macro

## Evaluation on the Data Cleaning

In [ ]:
## Data Imputation Inference

In [199]:
result

instruction  input  \
0     You are an expert in Cleaning Rayyan Dataset. ...    NaN   
1     You are an expert in Cleaning Rayyan Dataset. ...    NaN   
2     You are an expert in Cleaning Rayyan Dataset. ...    NaN   
3     You are an expert in Cleaning Rayyan Dataset. ...    NaN   
4     You are an expert in Cleaning Rayyan Dataset. ...    NaN   
...                                                 ...    ...   
1112  You are an expert in Cleaning Rayyan Dataset. ...    NaN   
1113  You are an expert in Cleaning Rayyan Dataset. ...    NaN   
1114  You are an expert in Cleaning Rayyan Dataset. ...    NaN   
1115  You are an expert in Cleaning Rayyan Dataset. ...    NaN   
1116  You are an expert in Cleaning Rayyan Dataset. ...    NaN   

                                   output                              predict  
0       {"article_jcreated_at": "1/1/71"}   {"article_jcreated_at": "1/13/00"}  
1               {"article_jvolumn": "-1"}            {"article_jvolumn": "-1"}  
2                {"article_jissue": "-1"}             {"article_jissue": "-1"}  
3      {"article_jcreated_at": "2/15/04"}   {"article_jcreated_at": "1/13/04"}  
4       {"article_jcreated_at": "1/6/12"}   {"article_jcreated_at": "1/13/04"}  
...                                   ...                                  ...  
1112            {"article_jvolumn": "-1"}            {"article_jvolumn": "-1"}  
1113             {"article_jissue": "-1"}             {"article_jissue": "-1"}  
1114  {"article_jcreated_at": "10/27/14"}   {"article_jcreated_at": "1/13/12"}  
1115   {"article_jcreated_at": "1/11/01"}   {"article_jcreated_at": "1/13/12"}  
1116   {"article_jcreated_at": "1/12/01"}   {"article_jcreated_at": "1/12/01"}  

[1117 rows x 4 columns]

In [4]:
## Try to test recall on Rayyan Dataset
import json
# result = pd.read_csv('/data/home/wangys/LLaMA-Factory-main/inference_multi_experts/merge_experts_hospital|Expert-7/mistral-7b-rayyan-test-20.csv',index_col=0)
# result = pd.read_csv('/data/home/wangys/LLaMA-Factory-main/inference_MELD/Vicuna-33B|rayyan-train-MoE/vicuna-33b-data_rayyan_rayyan-test-20.csv',index_col=0)
result = pd.read_csv('/data/home/wangys/LLaMA-Factory-main/inference_multi_experts/merge_experts_hospital|Expert-7/mistral-7b-rayyan-test-20.csv',index_col=0)

result = pd.read_csv('/data/home/wangys/DataSelection-IF/inference_Transfer_ER/select--rayyan-test.csv',index_col=0)

print(len(result))

# result = pd.read_csv('inference_Transfer_ER/select--rayyan-test.csv',index_col=0)
rayyan_detector = np.load('/data/home/wangys/MoE-Example/DC/GEIL_Data/rayyan/detector/detector.npy')
rayyan_clean = pd.read_csv('/data/home/wangys/MoE-Example/DC/GEIL_Data/rayyan/original/clean.csv').fillna('')
rayyan_dirty = pd.read_csv('/data/home/wangys/MoE-Example/DC/GEIL_Data/rayyan/original/dirty.csv').fillna('')
def Str2Int(row):
    for index in range(11):
        temp = row[index]
        try:
            row[index] = str(int(temp))
        except:
            continue
    return row
rayyan_clean = rayyan_clean.apply(Str2Int,axis=1)
rayyan_dirty = rayyan_dirty.apply(Str2Int,axis=1)
count = 0
valid_count = 0
rayyan_correction = rayyan_dirty.copy()
import ast
for d in tqdm(np.argwhere(rayyan_detector==1)):
    i = d[0]
    j = d[1] + 1 ## Ignore Index
    try:
        predict = list(eval(result.iloc[count,-1]).values())[0]
        rayyan_correction.iloc[i,j] = predict
        valid_count += 1
    except:
        # print(result.iloc[count,-1])
        predict = result.iloc[count,-1]
        rayyan_correction.iloc[i,j] = predict
    count += 1
All_Data_Error = 0
All_Fixed_Error = 0
Correct_Fixed_Error = 0
clean = rayyan_clean.copy()
dirty = rayyan_dirty.copy()
correction = rayyan_correction.copy()
for i in tqdm(range(len(clean))):
# for i in tqdm(tax_error):
    for j in range(clean.shape[1]):
        dirty_cell = dirty.iloc[i,j]
        clean_cell = clean.iloc[i,j]
        correct_cell = correction.iloc[i,j]
        if(correct_cell!=dirty_cell):
            All_Fixed_Error += 1
        if(clean_cell!=dirty_cell):
            All_Data_Error += 1
            if(correct_cell==clean_cell or correct_cell in clean_cell):
                Correct_Fixed_Error += 1
Precision_hospital = Correct_Fixed_Error / All_Fixed_Error
Recall_hospital = Correct_Fixed_Error / All_Data_Error
F1_hospital = (2 * Precision_hospital * Recall_hospital) / (Precision_hospital + Recall_hospital)
Precision_hospital,Recall_hospital,F1_hospital

1117


/tmp/ipykernel_3567902/1158201429.py:17: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`
  temp = row[index]
/tmp/ipykernel_3567902/1158201429.py:19: FutureWarning: Series.__setitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To set a value by position, use `ser.iloc[pos] = value`
  row[index] = str(int(temp))
/tmp/ipykernel_3567902/1158201429.py:17: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`
  temp = row[index]
/tmp/ipykernel_3567902/1158201429.py:19: FutureWarning: Series.__setitem__ treating keys as positions is d

  0%|          | 0/1117 [00:00<?, ?it/s]

  0%|          | 0/1000 [00:00<?, ?it/s]

(0.7815884476534296, 0.9135021097046413, 0.8424124513618677)

77.80
80.21
82.55
81.64
79.58
77.79
77.16
80.00

In [81]:
All_Data_Error = 0
All_Fixed_Error = 0
Correct_Fixed_Error = 0
clean = rayyan_clean.copy()
dirty = rayyan_dirty.copy()
correction = rayyan_correction.copy()
for i in tqdm(range(len(clean))):
# for i in tqdm(tax_error):
    for j in range(clean.shape[1]):
        dirty_cell = dirty.iloc[i,j]
        clean_cell = clean.iloc[i,j]
        correct_cell = correction.iloc[i,j]
        if(correct_cell!=dirty_cell):
            All_Fixed_Error += 1
        if(clean_cell!=dirty_cell):
            All_Data_Error += 1
            if(correct_cell==clean_cell):
                Correct_Fixed_Error += 1
Precision_hospital = Correct_Fixed_Error / All_Fixed_Error
Recall_hospital = Correct_Fixed_Error / All_Data_Error
F1_hospital = (2 * Precision_hospital * Recall_hospital) / (Precision_hospital + Recall_hospital)
Precision_hospital,Recall_hospital,F1_hospital

  0%|          | 0/1000 [00:00<?, ?it/s]

(0.7815884476534296, 0.9135021097046413, 0.8424124513618677)

## Beers

In [161]:
# beer_result = pd.read_csv('/data/home/wangys/LLaMA-Factory-main/inference_baseline/models|Mixtral-sft/Mixtral-beer-test-ablation.csv',index_col=0) ## generation-ablation
beer_result = pd.read_csv('inference_Transfer_ER/select--beer-test.csv',index_col=0) 
len(beer_result)

3342

In [110]:
beer_result = pd.read_json('/data/home/wangys/LLaMA-Factory-main/data/beer/beer-test-20.json')
len(beer_result)

3364

In [162]:
count = 0
beer_clean = pd.read_csv('/data/home/wangys/MoE-Example/DC/GEIL_Data/beers/original/clean.csv').fillna('')
beer_dirty = pd.read_csv('/data/home/wangys/MoE-Example/DC/GEIL_Data/beers/original/dirty.csv').fillna('')
detector_beer = np.load('/data/home/wangys/MoE-Example/DC/GEIL_Data/beers/detector/detector.npy')
beer_dirty.columns = beer_clean.columns
def try_convert_to_int(row):
    for x,y in row.items():
        if(x in ['ounces','ibu']):
            try:
                row[x] = int(y)
            except:
                row[x] = y
    return row
beer_clean = beer_clean.apply(try_convert_to_int,axis=1).astype(str)
beer_dirty = beer_dirty.apply(try_convert_to_int,axis=1).astype(str)
# beer_result = pd.read_csv('/data/home/wangys/LLaMA-Factory-main/inference_baseline/models|Mixtral-sft/Mixtral-beer-test-ablation.csv',index_col=0) ## generation-ablation
beer_result = pd.read_csv('inference_Transfer_ER/--beer-test-20.csv',index_col=0) ## generation-ablation
beer_correction = beer_dirty.copy()
for d in np.argwhere(detector_beer==1):
    i = d[0] 
    j = d[1] + 2
    
    try:
        predict = list(eval(beer_result.iloc[count,-1]).values())[0]
        beer_correction.iloc[i,j] = predict
        count += 1
    except:
        print(count)
        count += 1
print(count,len(np.argwhere(detector_beer==1)),len(beer_result))

3364 3364 3364


In [163]:
All_Data_Error = 0
All_Fixed_Error = 0
Correct_Fixed_Error = 0
for i in range(len(beer_clean)):
    for j in range(11):
        dirty_cell = beer_dirty.iloc[i,j]
        clean_cell = beer_clean.iloc[i,j]
        correct_cell = beer_correction.iloc[i,j]
        if(correct_cell!=dirty_cell):
            All_Fixed_Error += 1
        if(clean_cell!=dirty_cell):
            All_Data_Error += 1
            if(correct_cell==clean_cell):
                Correct_Fixed_Error += 1
Precision_hospital = Correct_Fixed_Error / All_Fixed_Error
Recall_hospital = Correct_Fixed_Error / All_Data_Error
F1_hospital = (2 * Precision_hospital * Recall_hospital) / (Precision_hospital + Recall_hospital)
Precision_hospital,Recall_hospital,F1_hospital

(0.9749776586237712, 0.9749776586237712, 0.9749776586237712)

In [209]:
All_Data_Error,All_Fixed_Error,Correct_Fixed_Error

(3357, 3364, 1724)

In [79]:
# hospital_result = pd.read_csv('/data/home/wangys/LLaMA-Factory-main/inference_multi_experts/merge_experts_hospital|Expert-8/mistral-7b-hospital-test.csv',index_col=0) ## 

hospital_result = pd.read_csv('inference_Transfer_ER/select--hospital-test.csv',index_col=0) ## 
count = 0
hospital_clean = pd.read_csv('/data/home/wangys/MoE-Example/DC/GEIL_Data/hospital/original/clean.csv').astype(str)
hospital_dirty = pd.read_csv('/data/home/wangys/MoE-Example/DC/GEIL_Data/hospital/original/dirty.csv').astype(str)
hospital_dirty.columns = hospital_clean.columns
hospital_correction = hospital_dirty.copy()
hospital_detector = np.load('/data/home/wangys/MoE-Example/DC/GEIL_Data/hospital/detector/detector.npy').reshape((-1,20))
import ast
for d in np.argwhere(hospital_detector==1):
    i = d[0]
    j = d[1]
    try:
        predict = list(eval(hospital_result.iloc[count,-1]).values())[0]
        hospital_correction.iloc[i,j] = predict
        count += 1
    except:
        predict = hospital_result.iloc[count,-1]
        hospital_correction.iloc[i,j] = predict
        count += 1
print(count,len(hospital_result))
All_Data_Error = 0
All_Fixed_Error = 0
Correct_Fixed_Error = 0
clean = hospital_clean.copy()
dirty = hospital_dirty.copy()
correction = hospital_correction.copy()
for i in tqdm(range(len(clean))):
# for i in tqdm(tax_error):
    for j in range(clean.shape[1]):
        dirty_cell = dirty.iloc[i,j]
        clean_cell = clean.iloc[i,j]
        correct_cell = correction.iloc[i,j]
        if(correct_cell!=dirty_cell):
            All_Fixed_Error += 1
        if(clean_cell!=dirty_cell):
            All_Data_Error += 1
            if(correct_cell==clean_cell or correct_cell in clean_cell):
                Correct_Fixed_Error += 1
Precision_hospital = Correct_Fixed_Error / All_Fixed_Error
Recall_hospital = Correct_Fixed_Error / All_Data_Error
F1_hospital = (2 * Precision_hospital * Recall_hospital) / (Precision_hospital + Recall_hospital)
Precision_hospital,Recall_hospital,F1_hospital

508 508


  0%|          | 0/1000 [00:00<?, ?it/s]

(0.9684418145956607, 0.9646365422396856, 0.9665354330708662)

{'init_select': 33.383955240249634, 'init-train': 147.88934469223022, 'batch-division': 151.29011631011963, 'gradient-calculation': 181.62943506240845, 'IF-Score': 195.9503378868103, 'Final Selection': 196.09836435317993, 'Selection fine-tune': 507.5114097595215, 'Selection Inference': 551.6968641281128}
(0.9684418145956607, 0.9646365422396856, 0.9665354330708662) 1500 Hospital

{'init_select': 45.18479037284851, 'init-train': 187.85570907592773, 'batch-division': 192.36323857307434, 'gradient-calculation': 227.50458121299744, 'IF-Score': 242.43565607070923, 'Final Selection': 242.51967883110046, 'Selection fine-tune': 559.618061542511, 'Selection Inference': 641.3211114406586}  1500 Rayyan
(0.7815884476534296, 0.9135021097046413, 0.8424124513618677)

(0.9749776586237712, 0.9749776586237712, 0.9749776586237712) 1500 beer



## Data Imputation

0.8653846153846154 walmart-test
{'init_select': 39.47596573829651, 'init-train': 194.13445043563843, 'batch-division': 204.5651400089264, 'gradient-calculation': 242.23899912834167, 'IF-Score': 264.88237619400024, 'Final Selection': 264.9920480251312, 'Selection fine-tune': 669.8673028945923, 'Selection Inference': 703.3083307743073}

In [78]:
All_Data_Error = 0
All_Fixed_Error = 0
Correct_Fixed_Error = 0
clean = hospital_clean.copy()
dirty = hospital_dirty.copy()
correction = hospital_correction.copy()
for i in tqdm(range(len(clean))):
# for i in tqdm(tax_error):
    for j in range(clean.shape[1]):
        dirty_cell = dirty.iloc[i,j]
        clean_cell = clean.iloc[i,j]
        correct_cell = correction.iloc[i,j]
        if(correct_cell!=dirty_cell):
            All_Fixed_Error += 1
        if(clean_cell!=dirty_cell):
            All_Data_Error += 1
            if(correct_cell==clean_cell or correct_cell in clean_cell):
                Correct_Fixed_Error += 1
Precision_hospital = Correct_Fixed_Error / All_Fixed_Error
Recall_hospital = Correct_Fixed_Error / All_Data_Error
F1_hospital = (2 * Precision_hospital * Recall_hospital) / (Precision_hospital + Recall_hospital)
Precision_hospital,Recall_hospital,F1_hospital

  0%|          | 0/1000 [00:00<?, ?it/s]

(0.9684418145956607, 0.9646365422396856, 0.9665354330708662)

### Cross-Task and Cross-Domain


In [ ]:
/data/home/wangys/LLaMA-Factory-main/data/MoE/ER/amazon-google-test.json
/data/home/wangys/LLaMA-Factory-main/data/MoE/ER/semi-text-w-test-MoE.json
/data/home/wangys/LLaMA-Factory-main/data/MoE/DI/walmart_test_output_wide.json
/data/home/wangys/LLaMA-Factory-main/data/MoE/DI/restaurant_test_output_wide.json
/data/home/wangys/LLaMA-Factory-main/data/CTA/WebTable_Test_few.json
/data/home/wangys/LLaMA-Factory-main/data/CTA/SimTab_test_few.json

In [243]:
list(MoE_list_update_top_2['domain'].unique())

['Mistral|amazon_google-MoE-CT',
 'Mistral|semi_text_w-MoE-CT',
 'Mistral|walmart-MoE-CT',
 'Mistral|restaurant-MoE-CT',
 'Mistral|webtable-MoE-CT',
 'Mistral|SimTab-MoE-CT']

In [224]:
MoE_list_update_top_2 = pd.read_csv('../MoE-Example/Router/MoE_list_update_top_2_cross_dataset.csv',index_col=0)

In [307]:
MoE_list_update_top_2 = pd.read_csv('../MoE-Example/Router/MoE_list_update_top_2_cross_task.csv',index_col=0)

In [308]:
# Amazon_Google_CD = MoE_list_update_top_2[MoE_list_update_top_2['domain']=='Mistral|amazon_google-MoE-CT']
# amazon_google_gt = pd.read_json('/data/home/wangys/LLaMA-Factory-main/data/MoE/ER/amazon-google-test.json')

# Amazon_Google_CD = MoE_list_update_top_2[MoE_list_update_top_2['domain']=='Mistral|semi_text_w-MoE-CT']
# amazon_google_gt = pd.read_json('/data/home/wangys/LLaMA-Factory-main/data/MoE/ER/semi-text-w-test-MoE.json')

# Amazon_Google_CD = MoE_list_update_top_2[MoE_list_update_top_2['domain']=='Mistral|SimTab-MoE-CT']
# amazon_google_gt = pd.read_json('/data/home/wangys/LLaMA-Factory-main/data/CTA/SimTab_test_few.json')

# Amazon_Google_CD = MoE_list_update_top_2[MoE_list_update_top_2['domain']=='Mistral|webtable-MoE-CT']
# amazon_google_gt = pd.read_json('/data/home/wangys/LLaMA-Factory-main/data/CTA/WebTable_Test_few.json')

# Amazon_Google_CD = MoE_list_update_top_2[MoE_list_update_top_2['domain']=='Mistral|walmart-MoE-CT']
# amazon_google_gt = pd.read_json('/data/home/wangys/LLaMA-Factory-main/data/MoE/DI/walmart_test_output_wide.json')

Amazon_Google_CD = MoE_list_update_top_2[MoE_list_update_top_2['domain']=='Mistral|restaurant-MoE-CT']
amazon_google_gt = pd.read_json('/data/home/wangys/LLaMA-Factory-main/data/MoE/DI/restaurant_test_output_wide.json')

In [309]:
amazon_google_gt_dict = dict(zip(amazon_google_gt['instruction'],amazon_google_gt['output']))
Amazon_Google_CD['output'] = Amazon_Google_CD['query'].map(amazon_google_gt_dict)

/tmp/ipykernel_1389237/3355500504.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  Amazon_Google_CD['output'] = Amazon_Google_CD['query'].map(amazon_google_gt_dict)


In [38]:
# result_merge = Amazon_Google_CD
result_merge = pd.read_csv('/data/home/wangys/LLaMA-Factory-main/inference_MoE/Mixtral-CT|CD-no-semi-text-w/Mixtral-7x8b-semi-text-w-test-MoE.csv',index_col=0)
def Transfer(row):
    if(row['output'].lower().__contains__('dismatch')):
        label = 0
    else:
        label = 1
    if(row['predict'].lower().__contains__('dismatch')):
        predict = 0
    else:
        predict = 1
    return label,predict
result_output = result_merge.apply(Transfer,axis=1,result_type='expand')
from sklearn.metrics import precision_score,recall_score,f1_score
print(precision_score(y_true=result_output[0],y_pred=result_output[1]),recall_score(y_true=result_output[0],y_pred=result_output[1]),f1_score(y_true=result_output[0],y_pred=result_output[1]))

0.38860103626943004 0.35545023696682465 0.3712871287128713


In [41]:
Amazon_Google_CD = pd.read_csv('/data/home/wangys/LLaMA-Factory-main/inference_MoE/Table-llama-CT|CD-no-webtable/table-llama-WebTable_Test_few.csv',index_col=0).fillna('')
count = 0
for index,row in Amazon_Google_CD.iterrows():
    try:
        truth = list(eval(row['output']).values())[0].lower().strip()
    except:
        truth = (row['output'])
    if(row['predict'].lower().strip().__contains__(truth)):
        count += 1
count / len(Amazon_Google_CD)

0.3892371110734655

In [ ]:
## Dataset - Expert
Walmart - Restaurant: Cross-Dataset with Top-1 expert 0.8461538461538461
Walmart - SimTable: 0.8076923076923077/0.8275862068965517

In [310]:
# Amazon_Google_CD
count = 0
for index,row in Amazon_Google_CD.iterrows():
    truth = list(eval(row['output']).values())[0].lower().strip()
    if(row['prediction'].lower().strip().__contains__(truth)):
        count += 1
count / len(Amazon_Google_CD)

0.7586206896551724

## Evaluation over one single data

In [16]:
file = pd.read_csv('/data/home/wangys/LLaMA-Factory-main/inference_noMeta/MoE-no-Meta|ant_buy_train-MoE-noMeta/mistral-7b-all_test.csv',index_col=0)

In [19]:
domain = file['domain'].unique()
for d in domain:
    select_df = file[file['domain']==d]
    result_merge = select_df
    def Transfer(row):
        if(row['output'].lower().__contains__('dismatch')):
            label = 0
        else:
            label = 1
        if(row['predict'].lower().__contains__('dismatch')):
            predict = 0
        else:
            predict = 1
        return label,predict
    result_output = result_merge.apply(Transfer,axis=1,result_type='expand')
    from sklearn.metrics import precision_score,recall_score,f1_score
    print(precision_score(y_true=result_output[0],y_pred=result_output[1]),recall_score(y_true=result_output[0],y_pred=result_output[1]),f1_score(y_true=result_output[0],y_pred=result_output[1]),d)

0.4726477024070022 0.9230769230769231 0.6251808972503619 amazon_google_test
0.28180039138943247 0.6824644549763034 0.39889196675900274 semi_text_w_test
0.5087538619979403 0.853195164075993 0.6374193548387096 semi_text_c_test
0.6964980544747081 0.927461139896373 0.7955555555555555 walmart_amazon_test
0.9267461669505963 0.9081803005008348 0.9173693086003373 wdc_all_test
0.8947368421052632 0.9077669902912622 0.9012048192771085 ant_buy_test


In [ ]:
'python vllm_inference_api_table_llama.py -checkpoint_dir lora_weight/Table-llama-CT/CD-no-SimTab -test_file /data/home/wangys/LLaMA-Factory-main/data/CTA/SimTab_test_few.json --count 0 --json',
'python vllm_inference_api_table_llama.py -checkpoint_dir lora_weight/Table-llama-CT/CD-no-webtable -test_file /data/home/wangys/LLaMA-Factory-main/data/CTA/WebTable_Test_few.json --count 0 --json',

In [ ]:
CUDA_VISIBLE_DEVICES=6,7 python vllm_inference_mistral_api.py --model_path /data/home/wangys/model/qwen-2.5-7B --directory lora/qwen2.5-7B/ER/amazon-google/select --gpu_num 2 --file train/ER/amazon-google/amazon-google-test.json --json --guided_choices

In [ ]:
CUDA_VISIBLE_DEVICES=6,7 python vllm_inference_mistral_api.py --model_path /data/home/wangys/model/qwen-2.5-7B  --gpu_num 2 --file train/ER/amazon-google/amazon-google-test.json --json --guided_choices